VERSIONING.md

```diff
@@ -0,0 +1,73 @@
+# OpenTelemetry Java Versioning
+
+## Compatibility requirements
+
+This codebase strictly follows [Semantic Versioning 2.0.0](https://semver.org/). This means
+that all artifacts have a version of the format `MAJOR.MINOR.PATCH` or `MAJOR.MINOR.PATCH-alpha`.
+For any artifact with a stable release, that is its version does not end in `-alpha`, no backwards-incompatible
+changes will be made unless incrementing the `MAJOR` version number. In practice, this means that
+backwards-incompatible changes will be avoided as long as possible. Most releases are made by
+incrementing the `MINOR` version. Patch releases with urgent cherry-picked bugfixes will be made by 
+incrementing the `PATCH` version.
+
+A backwards-incompatible change affects the public API of a module. The public API is any public
+class or method that is not in a package which includes the word `internal`. Examples of incompatible
+changes are:
+
+- API changes that could require code using the artifact to be changed, e.g., removing a method, 
+  reordering parameters, adding a method to an interface or abstract class without adding a default
+  implementation.
+  
+- ABI changes that could require code using the artifact to be recompiled, but not changed, e.g.,
```

Comment:
This is a great resource https://docs.oracle.com/javase/specs/jls/se7/html/jls-13.html

Reply:
Thanks, updated